addons/web/tests/test_res_users.py

```diff
@@ -17,19 +17,23 @@ def setUpClass(cls):
         ])
 
     def test_name_search(self):
-        """ Test name search with self assign feature """
+        """
+        Test name search with self assign feature
+        The self assign feature is present only when a limit is present,
+        which is the case with the public name_search by default
+        """
         ResUsers = self.env['res.users']
         jean = self.users[0]
-        user_ids = ResUsers.with_user(jean)._name_search('')
+        user_ids = [_id for _id, _ in ResUsers.with_user(jean).name_search('')]
```

Comment:
IMO, an underscore in front of a variable name means that you don't use that value. For `id` we usually use `id_` to avoid shadowing the builtin.
Also it's always better to use double _ if you don't care about the value to avoid Odoo's `_` shadowing.

Reply:
I knew the `id` had an `_` somewhere to avoid shadowing the builtin, guess I got the wrong side :P

> Also it's always better to use double _ if you don't care about the value to avoid Odoo's _ shadowing (from odoo import _). 

true that in odoo `_` is the translate function, not the pythonic 'discard' variable.